Description: Do not depend on zlib compression being available.
 The Debian OpenSSL package is built without support for zlib
 compression since version 1.0.1e-5.
Forwarded: not-yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2015-06-04

--- a/src/ssl.c
+++ b/src/ssl.c
@@ -120,11 +120,10 @@
         comp->method=COMP_zlib();
         if(!comp->method || comp->method->type==NID_undef) {
             OPENSSL_free(comp);
-            s_log(LOG_ERR, "Failed to initialize compression method");
-            return 1;
+        } else {
+            comp->name=(char *)(comp->method->name);
+            sk_SSL_COMP_push(ssl_comp_methods, comp);
         }
-        comp->name=(char *)(comp->method->name);
-        sk_SSL_COMP_push(ssl_comp_methods, comp);
     }
 
     /* also insert one of obsolete (ZLIB/RLE) algorithms */
@@ -140,9 +139,13 @@
         comp->id=0xe1; /* 225 - within private range (193 to 255) */
         comp->method=COMP_rle();
     } else {
-        s_log(LOG_INFO, "Compression enabled: %d algorithm(s)",
-            sk_SSL_COMP_num(ssl_comp_methods));
-        OPENSSL_free(comp);
+	const int count = sk_SSL_COMP_num(ssl_comp_methods);
+	OPENSSL_free(comp);
+	if(count==0) {
+	    s_log(LOG_ERR, "No valid compression algorithms found");
+	    return 1;
+	}
+        s_log(LOG_INFO, "Compression enabled: %d algorithm(s)", count);
         return 0;
     }
     if(!comp->method || comp->method->type==NID_undef) {
--- a/src/options.c
+++ b/src/options.c
@@ -439,9 +439,9 @@
         if(strcasecmp(opt, "compression"))
             break;
         if(SSLeay()>=0x00908051L && !strcasecmp(arg, "deflate"))
-            new_global_options.compression=COMP_DEFLATE;
+            return "deflate compression is not really implemented in OpenSSL";
         else if(!strcasecmp(arg, "zlib"))
-            new_global_options.compression=COMP_ZLIB;
+            return "zlib compression is disabled in the Debian package of OpenSSL";
         else if(!strcasecmp(arg, "rle"))
             new_global_options.compression=COMP_RLE;
         else
--- a/configure.ac
+++ b/configure.ac
@@ -179,10 +179,9 @@
 AC_SEARCH_LIBS([yp_get_default_domain], [nsl])
 AC_SEARCH_LIBS([socket], [socket])
 AC_SEARCH_LIBS([openpty], [util])
-# Checks for dynamic loader and zlib needed by OpenSSL
+# Checks for dynamic loader needed by OpenSSL
 AC_SEARCH_LIBS([dlopen], [dl])
 AC_SEARCH_LIBS([shl_load], [dld])
-AC_SEARCH_LIBS([inflateEnd], [z])
 
 # Add BeOS libraries
 if test "$host_os" = "beos"; then
